docs/source/tutorials/gateware_blinky.rst

```diff
@@ -0,0 +1,177 @@
+    :linenos:
+    :emphasize-lines: 7
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            leds: Signal(6) = Cat(platform.request("led", n).o for n in range(0, 6))
+
+            return m
+
+Amaranth platform resources can be obtained via the ``platform.request(name, number=0)`` method where ``name`` is the name of the resource and ``number`` is the index of the resource if there are more than one defined.
+
+In this case we use a Python list comprehension to obtain all six FPGA led's and concatenate them into a six-bit addressable Amaranth ``Signal`` using the ``Cat`` operation.
+
+
+Timer State
+===========
+
+To make the LED blink at predictable intervals we'll use a simple timer.
+
+To start with, let's define the timer state by adding the highlighted lines:
+
+.. code-block :: python
+    :linenos:
+    :emphasize-lines: 9-10
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            leds: Signal(6) = Cat(platform.request("led", n).o for n in range(0, 6))
+
+            half_freq: int    = int(60e6 // 2)
+            timer: Signal(25) = Signal(range(half_freq + 1))
+
+            return m
+
+First we'll declare a variable ``half_freq`` which is exactly half of Cynthion FPGA's default clock frequency in Hz, next we'll declare ``timer`` to be an Amaranth ``Signal`` which is wide enough to contain a value equal to ``half_freq + 1``.
+
+If we increment the ``timer`` by one for each clock cycle until it reaches ``half_freq`` we get a timer with a 500ms period.
+
+
+Timer Logic
+===========
+
+Now that we have a state definition for our timer we can move forward to the implementation logic, edit your file and add the highlighted lines:
+
+.. code-block :: python
+    :linenos:
+    :emphasize-lines: 12-17
+
+    from amaranth import *
+
+    class Top(Elaboratable):
+        def elaborate(self, platform):
+            m = Module()
+
+            leds: Signal(6) = Cat(platform.request("led", n).o for n in range(0, 6))
+
+            half_freq: int    = int(60e6 // 2)
+            timer: Signal(25) = Signal(range(half_freq))
+
+            with m.If(timer == half_freq - 1):
+                m.d.sync += leds.eq(~leds)
+                m.d.sync += timer.eq(0)
+
+            with m.Else():
+                m.d.sync += timer.eq(timer + 1)
+
+            return m
+
+Amaranth combines normal Python expressions with Amaranth in order to describe a design. Whenever you see the prefix ``m.`` you are making a call to the ``Module`` object you created at the beginning of the ``elaborate(...)`` method. These calls are what build the logic which makes up a design.
+
+The ``with m.If(...):`` and ``with m.Else():`` blocks operate much like you'd expect where, every clock-cycle, the expression ``timer == half_freq`` will be evaluated and trigger the corresponding branch.
```

Comment:
`timer == half_freq - 1`

Reply:
This is still pending

Reply:
I cannot even… I'm sorry @mndza! Fixed in: https://github.com/greatscottgadgets/cynthion/pull/197/commits/902ae987f31759142124bde53bb64ce0747a6068